strawberry/codemods/annotated_unions.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import annotations
+
+from typing import Sequence
+
+import libcst as cst
+import libcst.matchers as m
+from libcst._nodes.expression import BaseExpression, Call  # noqa: TCH002
+from libcst.codemod import CodemodContext, VisitorBasedCodemodCommand
+from libcst.codemod.visitors import AddImportsVisitor
+
+
+def _find_named_argument(args: Sequence[cst.Arg], name: str) -> cst.Arg | None:
+    return next(
+        (arg for arg in args if arg.keyword and arg.keyword.value == name),
+        None,
+    )
+
+
+def _find_positional_argument(
+    args: Sequence[cst.Arg], search_index: int
+) -> cst.Arg | None:
+    for index, arg in enumerate(args):
+        if index > search_index:
+            return None
+
+        if index == search_index and arg.keyword is None:
+            return arg
+
+
+class ConvertUnionToAnnotatedUnion(VisitorBasedCodemodCommand):
+    # TODO: support Union and | syntax, also in errors? ugh
+    DESCRIPTION: str = (
+        "Converts strawberry.union(..., types=(...)) to Annotated[Union[...]]"
+    )
+
+    def __init__(self, context: CodemodContext) -> None:
+        super().__init__(context)
+
+    @m.leave(m.Call(func=m.Attribute(value=m.Name("strawberry"), attr=m.Name("union"))))
+    def leave_union_call(
+        self, original_node: Call, updated_node: Call
+    ) -> BaseExpression:
+        types = _find_named_argument(original_node.args, "types")
+        union_name = _find_named_argument(original_node.args, "name")
```

Comment:
There's a pattern that avoids this kind of logic, which is using a `m.visit`/`m.leave` with `m.call_if_inside`. In other words, you could `@m.leave(m.Arg())`, making sure you are inside a:

```py
m.Call(func=m.Attribute(value=m.Name("strawberry"), attr=m.Name("union")))
```

Reply:
Or you could visit first, store variables inside `self.<something>`, and then on `m.leave` you'd consume and reset them.